### ISSUE TITLE
S3 bucket access

### ISSUE THREAD

**Issue by juangburgos** [jupyterlab-contrib/jupyterlab-filesystem-access#63]
Just want to start by saying _thank you_ for this extension, works flawlessly. IMO it should be bundled by default with jupyterlite, because it is such a basic functionality to be able to _persist the files outside of the browser_.

On the same lines, I'd like to propose a similar extension, but instead of using the Web Broswer File System Access API, use the [AWS Javascript SDK](https://docs.aws.amazon.com/sdk-for-javascript/v2/developer-guide/s3-examples.html) to persist files in the cloud. This would open up so many possibilities for remote work and sharing, plus now a days AWS is not the only S3 provider, which makes the S3 standard a universal language for cloud storage.

Thanks again.

**Comment by krassowski** [comment]
Like this one: https://github.com/jpmorganchase/jupyter-fs ?

**Comment by martinRenou** [comment]
I guess jupyter-fs requires a running server for it to work?

**Comment by juangburgos** [comment]
Yes, there are already jupyter extensions that allow to do what I proposed in a regular jupyter instance. But they are coded in python which in turn uses TCP sockets, which of course are not available in the web environment. If you try to use them in jupyterlite, you will get errors like:

```
ImportError: cannot import name 'ssl' from 'urllib3.util.ssl_' (/lib/python3.12/site-packages/urllib3/util/ssl_.py)
```

So this needs to be coded in javascript which can actually send the http requests required to comunicate with the cloud.
